webhook_service/main.go

```diff
@@ -0,0 +1,873 @@
+	go func() {
+		for {
+			results, err := rdb.BLPop(ctx, 0, inputQueue).Result()
+			if err != nil {
+				logger.Error(fmt.Sprintf("Input pop error: %v", err))
+				time.Sleep(time.Second)
+				continue
+			}
+
+			var msg Message
+			var rawStr string
+			rawMessage := []byte(results[1])
+			// This is stupid but we have to get the raw JSON string into a string before we can
+			// unmarshal it into our struct because of how it's encoded from CRCON
+			if err := json.Unmarshal(rawMessage, &rawStr); err == nil {
+				if err := json.Unmarshal([]byte(rawStr), &msg); err != nil {
+					logger.Error(fmt.Sprintf("unmarshal error: %v, JSON: %s", err, rawStr))
+					time.Sleep(time.Second)
+					continue
+				}
+			} else {
+				logger.Error(fmt.Sprintf("Input unmarshal error: %v, JSON: %s", err, rawMessage))
+				time.Sleep(time.Second)
+				continue
+			}
+
+			// Check to see if we already know this webhook's rate limit bucket
+			// We check by webhook ID since multiple webhooks can share the same bucket
+			webhookID, err := ExtractWebhookID(msg.Payload.URL)
+			if err != nil {
+				logger.Error(fmt.Sprintf("Could not parse webhook ID from: %s:%s", msg.Payload.URL, err))
+				continue
+			}
+			bucket, err := rdb.HGet(ctx, bucketsHash, webhookID).Result()
+			var msgBytes []byte
+			msgBytes, _ = json.Marshal(msg)
+			if err == redis.Nil {
+				// Unknown webhook, send to first-time queue
+				rdb.RPush(ctx, firstTimeQueue, msgBytes)
+				rdb.LTrim(ctx, firstTimeQueue, 0, maxQueueSize-1)
+			} else {
+				// Known bucket, send to bucket queue
+				queueKey := bucketQueuePrefix + bucket
+				rdb.RPush(ctx, queueKey, msgBytes)
+				rdb.LTrim(ctx, queueKey, 0, maxQueueSize-1)
+
+				// Start workers if new bucket
+				if _, exists := GetWorker(bucket, &mu); !exists {
+					worker := NewBucketWorker(bucket, rdb)
+					SetWorker(bucket, worker, &mu)
+					go worker.ProcessQueue(rdb)
+				}
+			}
+		}
+	}()
+
+	go ProcessFirstTime(rdb, &mu)
+	go SubscribeTransients(rdb, &mu)
+
+	// Keep running
+	select {}
+}
+
+// SetupRedis initializes the Redis client
+func SetupRedis() *redis.Client {
+	logger.Info("Initializing redis")
+	host := os.Getenv("HLL_REDIS_HOST")
+	if host == "" {
+		panic("HLL_REDIS_HOST not set")
+	}
+
+	portStr := os.Getenv("HLL_REDIS_PORT")
+	port, err := strconv.Atoi(portStr)
+	if err != nil || port <= 0 {
+		panic("HLL_REDIS_PORT not set or invalid")
+	}
+
+	addr := fmt.Sprintf("%s:%d", host, port)
+	return redis.NewClient(&redis.Options{
+		Addr: addr,
+		DB:   0, // CRCON uses DB #0 as a global shared database for all game servers
+	})
+}
+
+func ExtractWebhookID(url string) (string, error) {
+	matches := webhookIDPattern.FindStringSubmatch(url)
+	if len(matches) < 2 {
+		return "", fmt.Errorf("no webhook ID found in URL")
+	}
+	return matches[1], nil
+}
+
+func SetGloballyRateLimited(rdb *redis.Client, resetTime time.Time, resetAfter time.Duration) {
+	ctx := context.Background()
+	val := strconv.FormatInt(resetTime.Unix(), 10)
```

Comment:
`strconv.Itoa` is probably easier to understand then FormatInt in this simple case :)

Reply:
It could then also be inlined.